### ISSUE TITLE
[BUG] Server expecting true/false instead of True/False as Boolean parameter

### ISSUE THREAD

**Issue by DavidFarago** [schemathesis/schemathesis#1166]
**Describe the bug**
Schemathesis generates requests containing parameters like `is_last=True`, but the server responds with some 400 and logs
```
::ffff:127.0.0.1 - - [12/May/2021:15:01:44 +0000] "GET /protected/customers?is_last=True&name=%3F%1B%C3%BA&sort=%C3%AF%C3%A7%28%C2%AB&order=R%C2%A3%C3%87 HTTP/1.1" 400 67 "-" "schemathesis/3.6.4"
request.query.is_last should be boolean
```
I guess Python and Javascript do not understand each other for Boolean parameters. Boolean within JSON bodies do not cause problems.
 
So not only many valued logic break "tertium non datur" now ;)

**To Reproduce**
1. Develop a server in Javascript with `express-openapi-validator` with an endpoint containing a Boolean parameter
2. See error: requests get rejected with a 400 instead of processed correctly by the server.

The corresponding `openapi.yaml` contains
```yaml
        - name: last_minute
          in: query
          schema:
            type: boolean
```

As workaround, I switched in `openapi.yaml` for Boolean parameters (but not for body) to 
```
        - name: last_minute
          in: query
          schema:
            type: string
            format: jsBoolean
``` 
and used `schemathesis.register_string_format("jsBoolean", st.sampled_from(['true', 'false']))`. But this is a bit ugly, especially to have two different types for parameters vs. 

**Expected behavior**
Python and Javascript understanding each other for Boolean parameters, too -- either by setting a parameter or by allowing generators not only for string but also for other types (ideally for `type`s directly, without needing `format`s).

**Environment (please complete the following information):**
 - OS: Ubuntu 19.04
 - "express-openapi-validator": "^4.12.6",
 - Python version: 3.8.8
 - pytest 6.2.3
 - Schemathesis version: 3.6.4
 - Spec version: 3.0.0

**Comment by Stranger6667** [comment]
Hi @DavidFarago 

Thank you for reporting it! 

At the moment, there are no specific serialization rules for values in `header` / ` cookie` / `path` and `query` parameters (`body` has a separate serialization layer). Python-specific objects (e.g., booleans, None, etc.) are formatted as-is. From what I see in popular Python frameworks, `true` / `false` are usually accepted (e.g., in Django-filters or Flask-Rest-X), so I think that it is a much safer default at the moment. 

There were times when, e.g., Flask-Rest-Plus accepted only `True` / `False` and `1` / `0`, but it was quite long ago. Therefore I think that Schemathesis can just switch to `true` / `false` / `null`.

> by allowing generators not only for string but also for other types (ideally for types directly, without needing formats)

Interesting idea! For `body` it should be covered with [serializers](https://schemathesis.readthedocs.io/en/stable/how.html#payload-serialization), maybe something similar could be done for other locations. Changing it on the `type` level is quite problematic as the underlying `hypothesis-jsonschema` library doesn't support it as it follows the JSON Schema semantic there (where `format` is explicitly allowed to be defined by the end user)

**Comment by Stranger6667** [comment]
I think that I can take a deeper look at it next week :)

**Comment by DavidFarago** [comment]
Maybe the following is an interesting read with regard to generators for all basic types: https://www.researchgate.net/publication/221617639_JMLUnit_The_next_generation about JMLUnit, a tool similar to PBT where the user defines generators for basic types and the tool automatically generates complex objects via ctors with generated arguments.

**Comment by Stranger6667** [comment]
Thank you for sharing this :)

**Comment by Stranger6667** [comment]
Released in `3.8.0`
